packages/plugin-hardhat/src/utils/contract-instance.ts

```diff
@@ -24,23 +25,29 @@ export function getContractInstance(
   opts: DeployOpts & Platform,
   deployment: Deployment & DeployTransaction & RemoteDeploymentId,
 ) {
-  const instance = contract.attach(deployment.address);
+  const instance = attach(contract, deployment.address);
 
   // @ts-ignore Won't be readonly because instance was created through attach.
-  instance.deployTransaction = deployment.deployTransaction;
+  instance.deploymentTransaction = () => deployment.deployTransaction;
```

Comment:
Here we would have to stick to `null` to conform to Ethers' types.

```suggestion
  instance.deploymentTransaction = () => deployment.deployTransaction ?? null;
```